### PR TITLE
8221871: javadoc should not set role=region on <section> elements

### PR DESCRIPTION
**Pull request by psoujany**
Backport javadoc should not set role=region on <section> elements. 

The bug report for the same : https://bugs.openjdk.org/browse/JDK-8221871

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8221871](https://bugs.openjdk.org/browse/JDK-8221871): javadoc should not set role=region on <section> elements

### Reviewers
 * [Christoph Langer](https://openjdk.org/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk11u-dev pull/1219/head:pull/1219` \
`$ git checkout pull/1219`

Update a local copy of the PR: \
`$ git checkout pull/1219` \
`$ git pull https://git.openjdk.org/jdk11u-dev pull/1219/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1219`

View PR using the GUI difftool: \
`$ git pr show -t 1219`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk11u-dev/pull/1219.diff">https://git.openjdk.org/jdk11u-dev/pull/1219.diff</a>

</details>

### CODE REVIEW

File: test/langtools/jdk/javadoc/doclet/testHtmlVersion/TestHtmlVersion.java
Comment (RealCLanger) [inline test/langtools/jdk/javadoc/doclet/testHtmlVersion/TestHtmlVersion.java:27]: Please only add 8221871 here. The other two added items haven't been backported obviously.

**Review by RealCLanger — CHANGES_REQUESTED** [review]
I have one small remark. I'll run it through SAP's nightlies.

File: test/langtools/jdk/javadoc/doclet/testHtmlVersion/TestHtmlVersion.java
Comment (psoujany) [reply test/langtools/jdk/javadoc/doclet/testHtmlVersion/TestHtmlVersion.java]: Sure, made the changes. Please review. Thank you.

**Comment by RealCLanger** [comment]
> @psoujany Please do not rebase or force-push to an active PR as it invalidates existing review comments. All changes will be squashed into a single commit automatically when integrating. See [OpenJDK Developers’ Guide](https://openjdk.org/guide/#working-with-pull-requests) for more information.

Yep, in that case an incremental commit would have been nicer since your first version has already been reviewed.

**Comment by psoujany** [comment]
> > @psoujany Please do not rebase or force-push to an active PR as it invalidates existing review comments. All changes will be squashed into a single commit automatically when integrating. See [OpenJDK Developers’ Guide](https://openjdk.org/guide/#working-with-pull-requests) for more information.
> 
> Yep, in that case an incremental commit would have been nicer since your first version has already been reviewed.

Do I need to revert my rebase now? Please let me know I'll try to do that.

**Comment by RealCLanger** [comment]
> Do I need to revert my rebase now? Please let me know I'll try to do that.

No, it's ok. Just for the next time...

**Comment by RealCLanger** [comment]
Our testing shows failures in jdk/javadoc/doclet/testHtmlVersion/TestHtmlVersion.java, please have a look.

**Comment by RealCLanger** [comment]
> > Our testing shows failures in jdk/javadoc/doclet/testHtmlVersion/TestHtmlVersion.java, please have a look.
> 
> Fixed the test failures, Could you please check once. Thank you.

I've reenabled this patch in our CI but too late for tonight's run. So we'll have to wait another 24 hours to see results.

There are also whitespace issues reported by jcheck. Please address them.

**Review by RealCLanger — APPROVED** [review]
Looks good now. SAP regression testing shows no problems any more.

**Comment by RealCLanger** [comment]
@psoujany You must not integrate a fix in a JDK Updates project before requesting (and receiving) maintainer approval. Please refer to the [Wiki](https://wiki.openjdk.org/display/JDKUpdates/How+to+contribute+a+fix), especially point 6.

In case you don't have JBS access, I can add the fix request on your behalf. In that case, please post the text for the fix request comment here and I'll copy&paste it over. Thx.

**Comment by psoujany** [comment]
@RealCLanger Apologies for adding integrate tag. I don't have access to JBS. Please add _Fix Request 11_. This is actual openjdk bug https://bugs.openjdk.org/browse/JDK-8221871

**Comment by RealCLanger** [comment]
> @RealCLanger Apologies for adding integrate tag. I don't have access to JBS. Please add _Fix Request 11_. This is actual openjdk bug https://bugs.openjdk.org/browse/JDK-8221871. Thank you.

I've added the fix request metadata to the JBS bug. Let's wait for approval.

**Comment by GoeLin** [comment]
Hi @psoujany, 
next time please closely follow the backport instructions.https://wiki.openjdk.org/display/JDKUpdates/How+to+contribute+a+fix  If you don't understand them, ask before acting.
* Test your change. GHA is only a final check!
* Tell what you have tested
* State what you resolved if it is not clean
* Don't integrate if you don't have the tag.
I'll apporve the change now.

**Comment by psoujany** [comment]
Sure, Thank you.

File: test/langtools/jdk/javadoc/doclet/testHtmlVersion/TestHtmlVersion.java
Comment (pavelrappo) [inline test/langtools/jdk/javadoc/doclet/testHtmlVersion/TestHtmlVersion.java:163]: I recently backported the original change JDK-8221871 to some other release and noticed this diff. I wonder why these lines were deleted? They seem related to JDK-8215599 which hasn't been backported.

Comment (psoujany) [reply test/langtools/jdk/javadoc/doclet/testHtmlVersion/TestHtmlVersion.java:163]: Hi, while backporting JDK-8221871 I've seen test failures at this code hence removed it which passed the tests. I wasn't aware JDK-8215599 needs to be backported first.

Comment (pavelrappo) [reply test/langtools/jdk/javadoc/doclet/testHtmlVersion/TestHtmlVersion.java:163]: I cannot see that JDK-8215599 has been ever backported, which is fine; I didn't say it should. However, I somehow managed to make those deleted lines work by accounting for the change brought by JDK-8221871 and was simply wondering what happened here. Here's the ammended change:
```
        // Test for package-frame page
        checkOutput("pkg/package-frame.html", true,
                "<!DOCTYPE HTML>",
                "<meta name=\"dc.created\"",
                "<main role=\"main\">\n"
                + "<h1 class=\"bar\"><a href=\"package-summary.html\" target=\"classFrame\">pkg</a></h1>",
                "<section>\n"
                + "<h2 title=\"Interfaces\">Interfaces</h2>",
                "<section>\n"
                + "<h2 title=\"Classes\">Classes</h2>",
                "<section>\n"
                + "<h2 title=\"Enums\">Enums</h2>",
                "<section>\n"
                + "<h2 title=\"Exceptions\">Exceptions</h2>",
                "<section>\n"
                + "<h2 title=\"Errors\">Errors</h2>",
                "<section>\n"
                + "<h2 title=\"Annotation Types\">Annotation Types</h2>");
```

Comment (pavelrappo) [reply test/langtools/jdk/javadoc/doclet/testHtmlVersion/TestHtmlVersion.java:163]: > Hi, while backporting JDK-8221871 I've seen test failures at this code hence removed it which passed the tests. I wasn't aware JDK-8215599 needs to be backported first.

Thanks for clarification.

**Pull request closed without merging**
